scipy/cluster/hierarchy.py

```diff
@@ -3358,9 +3362,23 @@ def llf(id):
                          ax=ax,
                          above_threshold_color=above_threshold_color)
 
+    R["leaves_color_list"] = _get_leaves_color_list(R)
+
     return R
 
 
+def _get_leaves_color_list(R):
+    leaves_color_list = [None] * len(R['leaves'])
+    for link_x, link_y, link_color in zip(R['icoord'],
+                                          R['dcoord'],
+                                          R['color_list']):
+        for (xi, yi) in zip(link_x, link_y):
+            if yi == 0.0:  # if yi is 0.0, the point is a leaf
+                # xi is 5, 15, 25, 35, ...
```

Comment:
This confused me. It seems to be related to the hardcoded x-axis values in `iv_ticks`. Could you expand the comment a bit to make clear what's going on here?

Reply:
Thank you for reviewing. I added additional comments. PTAL